cookbook/case_studies/feature_engineering/feast_integration/feast_workflow.py

```diff
@@ -0,0 +1,179 @@
+
+    config = RepoConfig(
+        registry=registry.remote_source,
+        project=f"horsecolic",
+        # Notice the use of a custom provider.
+        provider="custom_provider.provider.FlyteCustomProvider",
+        offline_store=FileOfflineStoreConfig(),
+        online_store=SqliteOnlineStoreConfig(),
+    )
+    return FeatureStore(config=config)
+
+
+sql_task = SQLite3Task(
+    name="sqlite3.horse_colic",
+    query_template="select * from data",
+    output_schema_type=FlyteSchema,
+    task_config=SQLite3Config(
+        uri=DATABASE_URI,
+        compressed=True,
+    ),
+)
+
+@task
+def store_offline(registry: FlyteFile, dataframe: FlyteSchema) -> (FlyteFile, str):
+    horse_colic_entity = Entity(name="Hospital Number", value_type=ValueType.STRING)
+
+    horse_colic_feature_view = FeatureView(
+        name="horse_colic_stats",
+        entities=["Hospital Number"],
+        features=[
+            Feature(name="rectal temperature", dtype=ValueType.FLOAT),
+            Feature(name="total protein", dtype=ValueType.FLOAT),
+            Feature(name="peripheral pulse", dtype=ValueType.FLOAT),
+            Feature(name="surgical lesion", dtype=ValueType.STRING),
+            Feature(name="abdominal distension", dtype=ValueType.FLOAT),
+            Feature(name="nasogastric tube", dtype=ValueType.STRING),
+            Feature(name="outcome", dtype=ValueType.STRING),
+            Feature(name="packed cell volume", dtype=ValueType.FLOAT),
+            Feature(name="nasogastric reflux PH", dtype=ValueType.FLOAT),
+        ],
+        batch_source=FileSource(
+            path=str(dataframe.remote_path),
+            event_timestamp_column="timestamp",
+        ),
+        ttl=timedelta(days=1),
+    )
+
+    fs = _build_feature_store(registry=registry)
+
+    # Ingest the data into feast
+    fs.apply([horse_colic_entity, horse_colic_feature_view])
+
+    return FlyteFile(registry.remote_source), horse_colic_feature_view.name
+
+@task
+def load_historical_features(registry: FlyteFile) -> FlyteSchema:
+    entity_df = pd.DataFrame.from_dict(
+        {
+            "Hospital Number": [
+                "530101",
+                "5290409",
+                "5291329",
+                "530051",
+                "529518",
+                "530101",
+                "529340",
+                "5290409",
+                "530034",
+            ],
+            "event_timestamp": [
+                datetime(2021, 6, 25, 16, 36, 27),
+                datetime(2021, 6, 25, 16, 36, 27),
+                datetime(2021, 6, 25, 16, 36, 27),
+                datetime(2021, 6, 25, 16, 36, 27),
+                datetime(2021, 6, 25, 16, 36, 27),
+                datetime(2021, 7, 5, 11, 36, 1),
+                datetime(2021, 6, 25, 16, 36, 27),
+                datetime(2021, 7, 5, 11, 50, 40),
+                datetime(2021, 6, 25, 16, 36, 27),
+            ],
+        }
+    )
+
+    fs = _build_feature_store(registry=registry)
```

Comment:
i still think we should make FeatureStore a type and it can be represented as a 
FlyteFile + Metadata?

Reply:
agreed, that's what I meant as the second TODO item in the PR description. Will do that in the next batch of updates.